tests/integration/suite/actors/reminders/serialization_json.go

```diff
@@ -0,0 +1,105 @@
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/sqlite"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(serializationJSON))
+}
+
+// serializationJSON tests:
+// - That reminders are serialized to JSON when the Actors API level in the cluster is < 20
+type serializationJSON struct {
+	daprd   *daprd.Daprd
+	srv     *prochttp.HTTP
+	handler *serializationHTTPServer
+	place   *placement.Placement
+	db      *sqlite.SQLite
+}
+
+func (i *serializationJSON) Setup(t *testing.T) []framework.Option {
+	// Init placement with a maximum API level of 10
+	i.place = placement.New(t,
+		placement.WithMaxAPILevel(10),
+	)
+
+	// Create a SQLite database
+	i.db = sqlite.New(t, sqlite.WithActorStateStore(true))
+
+	// Init daprd and the HTTP server
+	i.handler = &serializationHTTPServer{}
+	i.srv = prochttp.New(t, prochttp.WithHandler(i.handler.NewHandler()))
+	i.daprd = daprd.New(t,
+		daprd.WithResourceFiles(i.db.GetComponent()),
+		daprd.WithPlacementAddresses("localhost:"+strconv.Itoa(i.place.Port())),
+		daprd.WithAppPort(i.srv.Port()),
+		// Daprd is super noisy in debug mode when connecting to placement.
+		daprd.WithLogLevel("info"),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(i.db, i.place, i.srv, i.daprd),
+	}
+}
+
+func (i *serializationJSON) Run(t *testing.T, ctx context.Context) {
+	// Wait for placement to be ready
+	i.place.WaitUntilRunning(t, ctx)
+
+	// Wait for daprd to be ready
+	i.daprd.WaitUntilRunning(t, ctx)
+
+	// Wait for actors to be ready
+	err := i.handler.WaitForActorsReady(ctx)
+	require.NoError(t, err)
+
+	client := util.HTTPClient(t)
+	baseURL := fmt.Sprintf("http://localhost:%d/v1.0/actors/myactortype/myactorid", i.daprd.HTTPPort())
+
+	// Invoke an actor to confirm everything is ready to go
+	serializationInvokeActor(t, ctx, baseURL, client)
+
+	// Store a reminder
+	// This causes the data in the state store to be updated
+	serializationStoreReminder(t, ctx, baseURL, client)
+
+	// Check the data in the SQLite database
+	// The value must begin with `[{`, which indicates it was serialized as JSON
+	storedVal := serializationLoadRemindersFromDB(t, ctx, i.db.GetConnection())
+	assert.Truef(t, strings.HasPrefix(storedVal, "[{"), "Prefix not found in value: '%v'", storedVal)
+
+	// Ensure the reminder was invoked at least once
+	assert.Eventually(t, func() bool {
+		return i.handler.remindersInvokeCount.Load() > 0
+	}, time.Second, 10*time.Millisecond, "Reminder was not invoked at least once")
```

Comment:
A second seems very low, are we worried about this?

Reply:
```suggestion
	}, 5*time.Second, 10*time.Millisecond, "Reminder was not invoked at least once")
```